var/spack/repos/builtin/packages/highfive/package.py

```diff
@@ -45,8 +45,9 @@ class Highfive(CMakePackage):
     variant('mpi', default=True, description='Support MPI')
 
     depends_on('boost @1.41:', when='+boost')
-    depends_on('hdf5 ~mpi', when='~mpi')
-    depends_on('hdf5 +mpi', when='+mpi')
+    depends_on('hdf5',         when='~mpi')
```

Comment:
That is actually the central change behind "Ability for HighFive and SynapseTool to build non-mpi versions linked to parallel-hdf5". This is working, however I have my doubts that spack works well in all cases, instead I believe it assumes the default (+mpi) instead of finding the concrete plan which matches